test/e2e/cluster/add_remove_cluster.go

```diff
@@ -18,29 +18,29 @@ var _ = Describe("[expansion] Test expanding/shrinking an existing cluster fleet
 		clusterBName := framework.TestContext.KubeContexts[framework.ClusterB]
 		clusterCName := framework.TestContext.KubeContexts[framework.ClusterC]
 
+		gatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, true)
+		Expect(gatewayNode).To(HaveLen(0), fmt.Sprintf("Expected no gateway node on %q", framework.ClusterC))
+
 		By(fmt.Sprintf("Verifying that a pod in cluster %q cannot connect to a pod in cluster %q", clusterAName, clusterCName))
 		dataplane.RunNoConnectivityTest(f, false, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterA, framework.ClusterC)
 
 		By(fmt.Sprintf("Verifying that a pod in cluster %q cannot connect to a service in cluster %q", clusterBName, clusterCName))
-		dataplane.RunNoConnectivityTest(f, true, framework.GatewayNode, framework.NonGatewayNode, framework.ClusterB, framework.ClusterC)
+		dataplane.RunNoConnectivityTest(f, true, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterB, framework.ClusterC)
 
-		nonGatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, false)[0].Name
+		nonGatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, false)
+		Expect(nonGatewayNode).To(HaveLen(1), fmt.Sprintf("Found non-gateway node %q on %q", nonGatewayNode[0].Name, clusterCName))
 		By(fmt.Sprintf("Adding cluster %q by setting the gateway label on node %q", clusterCName, nonGatewayNode))
-		f.SetGatewayLabelOnNode(framework.ClusterC, nonGatewayNode, true)
-		gatewayNodes := f.FindNodesByGatewayLabel(framework.ClusterC, true)
-		Expect(gatewayNodes).To(HaveLen(1), fmt.Sprintf("Expected only one gateway node on %q", framework.ClusterC))
+		f.SetGatewayLabelOnNode(framework.ClusterC, nonGatewayNode[0].Name, true)
 
 		enginePod := f.AwaitSubmarinerEnginePod(framework.ClusterC)
 		By(fmt.Sprintf("Found submariner engine pod %q on %q", enginePod.Name, clusterCName))
 
 		By(fmt.Sprintf("Checking connectivity between clusters"))
-		dataplane.RunConnectivityTest(f, false, framework.GatewayNode, framework.GatewayNode, framework.ClusterC, framework.ClusterB)
-		dataplane.RunConnectivityTest(f, true, framework.GatewayNode, framework.GatewayNode, framework.ClusterC, framework.ClusterA)
+		dataplane.RunConnectivityTest(f, false, framework.GatewayNode, framework.GatewayNode, framework.ClusterB, framework.ClusterC)
+		dataplane.RunConnectivityTest(f, true, framework.GatewayNode, framework.GatewayNode, framework.ClusterA, framework.ClusterC)
```

Comment:
Why did you swap these? We want the listener pod on ClusterC.

Reply:
Done